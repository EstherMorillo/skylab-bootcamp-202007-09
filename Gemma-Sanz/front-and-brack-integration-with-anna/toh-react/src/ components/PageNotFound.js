import React from 'react';
import { Link, Redirect } from 'react-router-dom';

function PageNotFound() {
	const condition = false;
	return (
		<section>
			{condition === true && <Redirect to="/" />}
			<h1>Error - 404</h1>
			<p>Nothing to see here!</p>
			<Link to="/">Go back to Dasboard</Link>
		</section>
	);
}

export default PageNotFound;
