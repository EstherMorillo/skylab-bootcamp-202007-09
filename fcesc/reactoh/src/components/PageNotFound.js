import React from 'react';
import { NavLink } from 'react-router-dom';
import './../css/pageNotFound.css';
import * as ROUTES from './../config/routes';

class PageNotFound extends React.Component{
    
    getResult(){
        //<NavLink to={()=>{this.props.history.goBack()}}>Go back</NavLink>
        let p = (
            <div className="404__container">
                <h2>Error 404</h2>
                <p>Página no encontrada.</p>
                <NavLink to={ROUTES.HOME}>Página principal</NavLink>
            </div>
        )
        return p;
    }

    render(){
        return this.getResult();
    }
}

export default PageNotFound;