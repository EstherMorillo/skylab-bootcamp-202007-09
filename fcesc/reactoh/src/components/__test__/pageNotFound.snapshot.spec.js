import React from 'react';
import renderer from 'react-test-renderer';
import PageNotFound from './../PageNotFound';
import { BrowserRouter as Router } from 'react-router-dom';

const tree = renderer.create(<Router><PageNotFound /></Router>).toJSON();

describe('PageNotFound snapshot', () => {
    test('should match', () => {
        expect(tree).toMatchSnapshot();
    })
})
