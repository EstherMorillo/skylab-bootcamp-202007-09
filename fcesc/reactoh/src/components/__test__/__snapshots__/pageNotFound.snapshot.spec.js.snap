// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound snapshot should match 1`] = `
<div
  className="404__container"
>
  <h2>
    Error 404
  </h2>
  <p>
    Página no encontrada.
  </p>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Página principal
  </a>
</div>
`;
